src/units/rai.rs

```diff
@@ -36,9 +36,9 @@ impl Rai {
         Self(v.into())
     }
 
-    pub fn from_hex(s: &str) -> anyhow::Result<Self> {
+    pub fn from_hex(s: &str) -> Result<Self, FeelessError> {
         expect_len(s.len(), Rai::LEN * 2, "Hex rai")?;
-        let vec = hex::decode(s.as_bytes()).context("Decoding hex rai")?;
```

Comment:
Some context is lost here. If a call is made and there's multiple `hex::decode` calls, the caller/user doesn't know which one had the error. Not sure what the best way to deal with it is. Thoughts?